src/lib/components/chat/Chat.svelte

```diff
@@ -132,15 +132,33 @@
 			console.log(data);
 			let message = history.messages[data.message_id];
 
-			const status = {
-				done: data?.data?.done ?? null,
-				description: data?.data?.status ?? null
-			};
+			const type = data?.data?.type ?? null;
+			if (type === "status") {
+				const status = {
+					done: data?.data?.done ?? null,
+					description: data?.data?.status ?? null
+				};
+
+				if (message.statusHistory) {
+					message.statusHistory.push(status);
+				} else {
+					message.statusHistory = [status];
+				}
+			} else if (type === "citation") {
+				console.log(data);
+				const citation = {
```

Comment:
```
{
        "type": "citation",
        "data": {
                        "document": doc["page_content"],
                        "metadata": doc["metadata"],
                        "source": doc["url"],
        }
}
```

Would be more appropriate here to make it more generalisable, the rest LGTM!

Reply:
Fixed:

```python
        await __event_emitter__(
            {
                "type": "status",
                "data": {"description": "Retrieving user data", "done": False},
            }
        )
        await __event_emitter__(
            {
                "type": "citation",
                "data": {
                    "document": ["My document content"],
                    "metadata": {"source": "https://openwebui.com"},
                    "source": {"name": "My source title"},
                },
            }
        )
        try:
            result = post(url, data=params, headers=headers)
            result.raise_for_status()
        except RequestException as e:
            return json.dumps({"content": str(e)})
        await __event_emitter__(
            {
                "type": "status",
                "data": {"description": "Retrieving user data", "done": True},
            }
        )    
```